plugins/modules/aci_l4l7_device_selection_device.py

```diff
@@ -0,0 +1,293 @@
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_l4l7_device_selection_device
+short_description: Manage L4-L7 Device Selection Policy Devices (vns:RsLDevCtxToLDev)
+description:
+- Manage L4-L7 Device Selection Policy Logical Device Bindings
+options:
+  tenant:
+    description:
+    - Name of an existing tenant.
+    type: str
+    aliases: [ tenant_name ]
+  contract:
+    description:
+    - Name of an existing contract
+    type: str
+    aliases: [ contract_name ]
+  graph:
+    description:
+    - Name of an existing Service Graph Template
+    type: str
+    aliases: [ service_graph, service_graph_name ]
+  node:
+    description:
+    - Name of an existing Service Graph Node
+    type: str
+    aliases: [ node_name ]
+  device:
+    description:
+    - Logical device name
+    type: str
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    type: str
+    choices: [ absent, present, query ]
+    default: present
+extends_documentation_fragment:
+- cisco.aci.aci
+
+notes:
+- The C(tenant) and C(device selection policy) must exist before using this module in your playbook.
+  The M(cisco.aci.aci_tenant) and M(cisco.aci.aci_l4l7_device_selection_policy) modules can be used for this.
+seealso:
+- module: aci_l4l7_service_graph_template
+- module: aci_l4l7_service_graph_template_node
+- module: aci_l4l7_device
+- module: aci_contract
+- name: APIC Management Information Model reference
+  description: More information about the internal APIC classes, B(vnsRsLDevCtxToLDev)
+  link: https://developer.cisco.com/docs/apic-mim-ref/
+author:
+- Tim Cragg (@timcragg)
+'''
+
+EXAMPLES = r'''
+- name: Add a new logical device to a device selection policy
+  cisco.aci.aci_l4l7_device_selection_device:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: my_tenant
+    contract: my_contract
+    graph: my_graph
+    node: my_node
+    device: my_device
+    state: present
+  delegate_to: localhost
+
+- name: Remove logical device from a device selection policy
+  cisco.aci.aci_l4l7_device_selection_device:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: my_tenant
+    contract: my_contract
+    graph: my_graph
+    node: my_node
+    device: my_device
+    state: absent
+  delegate_to: localhost
+
+- name: Query device binding for a device selection policy
+  cisco.aci.aci_l4l7_device_selection_device:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: my_tenant
+    contract: my_contract
+    graph: my_graph
+    node: my_node
+    state: query
+  delegate_to: localhost
+  register: query_result
+
```

Comment:
is it possible to query all, please add example

Reply:
query all does not work with this module